contracts/src/systems/transport/contracts/donkey_systems.cairo

```diff
@@ -46,15 +48,26 @@ mod donkey_systems {
             donkeys.save(ref world);
 
             // emit burn donkey event
+            let time = starknet::get_block_timestamp();
             world
                 .emit_event(
                     @BurnDonkey {
                         entity_id: payer_id,
                         player_address: starknet::get_caller_address(),
                         amount: donkey_amount,
-                        timestamp: starknet::get_block_timestamp()
+                        timestamp: time
                     }
                 );
+
+            // [Achievement] Consume donkeys
+            if donkey_amount != 0 {
+                let count = (donkey_amount % core::integer::BoundedU32::max().into()).try_into().unwrap();
```

Comment:
I would like to have an integer rounded down here (without the `decimals` precision)
I am not sure were to fetch this decimal from here, if you have an idea @credence0x

Reply:
to get the count without precision, it would be `let count = donkey_amount  / RESOURCE_PRECISION;`. the precision can be gotten from `eternum::constants::RESOURCE_PRECISION`. 

i think that could lead to an innacurate donkey burn count though. if we want it to be accurate, we should use the precision